grommet/grommet
# Fix Carousel Transition Bug

#### What does this PR do?
This fixes a regression introduced to the `Carousel` component that caused un-smooth and jarring transitions between Carousel slides. The issue is caused when the user changes the current slide, and the `visibility` property is set to `hidden` before the slide can complete its animation of fading out. Thus, the current child is immediately removed while the next child is slowly sliding in.
#### Where should the reviewer start?
The only changes were made in the `Carousel/Carousel.js` file.
#### What testing has been done on this PR?
The accessibility issue that this regression was introduced in is still fixed. All other functionality of the component is maintained.
#### How should this be manually tested?
Run storybook and see the Carousel stories. Make sure the accessibility issue that [this PR](https://github.com/grommet/grommet/pull/5326) fixed is still working. Also, ensure that the Carousel transitions between slides are smooth and as expected. Compare the transitions with what is on `master`.
#### What are the relevant issues?
#5675 

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
Yes
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible 👍
